accepted/00075-live-patching-enhancements.md

```diff
@@ -0,0 +1,170 @@
+ - Display EOL dates for the current live patch in system details
+ - Display an out-of-support systems list in the home page
+ - Use the notifications system and/or send emails to report live patched systems approaching/past EOL
+
+### Data Retrieval
+
+This section evaluates three alternative implementations for this feature.
+
+#### Call 'zypper lifecycle' on client
+
+Uyuni calls `zypper lifecycle` on the client and collects the data via Salt during package profile update. The data is then stored in the database per client along with the kernel live version data.
+
+**Pros:**
+ - `zypper lifecycle` is publicly supported
+ - Can be attached to profile update data with an existing process
+
+**Cons:**
+ - `zypper lifecycle` currently provides no machine-readable format
+ - Extra packages need to be installed on the client:
+   - `zypper-lifecycle-plugin`
+   - `lifecycle-data-sle-module-live-patching`
+
+#### Read the lifecycle data file on client
+
+Uyuni runs a Salt state to read the lifecycle data file provided with the package `lifecycle-data-sle-module-live-patching` on the client and collects the data during package profile update. The data is then stored in the database per client along with the kernel live version data.
+
+**Pros:**
+ - Easy to implement
+ - Can be attached to profile update data with an existing process
+
+**Cons:**
+ - File name or format can change in the future
+ - `lifecycle-data-sle-module-live-patching` package must be installed on the client
+
+#### Unpack and read the lifecycle data RPM on Uyuni server
+
+The lifecycle information is provided in the `lifecycle-data-sle-module-live-patching` package in the Live Patching product. Uyuni unpacks this RPM and reads the data from the extracted CSV file during reposync of this product. The data is stored in the database per live patch package.
+
+**Pros:**
+ - Independent, no impositions on clients
+ - Gathered data is always available and can be used for UI enhancements such as displaying EOL info for all live patch packages displayed in the UI
+
+**Cons:**
+ - RPM format can change in the future
+ - Uyuni doesn't have any existing functionality that can be utilized
+ - Unpacking must be done during reposync, adding complextion
+
+## 2. Live patching CLM workflow
+
+SLES channels can be prepared for live patching applications using the Content Lifecycle Management feature in Uyuni. This is done by adding a set of filters that exclude the regular kernel package updates and include live patch versions of those. However, the UI doesn't provide any special workflow for live patching and this process must be done manually by the user.
+
+Below are some examples of such sets of filters:
+
+Live patching project with a single filter to exclude regular kernel packages newer than a specific version:
+ - `DENY erratum contains package with epoch/version/release greater than kernel-default <EVR> (package_nevr)`
+
+Live patching project using `ALLOW` overrides:
```

Comment:
```suggestion
Alternative way for live patching project filters using `ALLOW` overrides:
```
Call it "alternative" to make it clear that this is not required together with the filter above.

Reply:
What would be pros and cons of the two alternative approaches? Is now (RFC time) the right time to decide which path to take?

Reply:
These are just meant to be 2 different examples to solve the problem with the current CLM manually.

> Is now (RFC time) the right time to decide which path to take?

I believe not. It's an implementation detail and can be shaped by testing different combinations.

Reply:
I think we should decide during the RFC process. AFAIK we documented option 1 in our manuals, while option 2 was choosen by @dvosburg . If we implement one option, but sales do not like it we do not win anything.